src/Microsoft.AspNet.Mvc.Core/ActionConstraints/ConsumesAttribute.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Mvc.HeaderValueAbstractions;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Specifies the allowed content types which can be used to help in selecting the action based on request's
+    /// content-type.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class ConsumesAttribute : Attribute, IResourceFilter, IConsumesActionConstraint
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="ConsumesAttribute"/>.
+        /// </summary>
+        public ConsumesAttribute(string contentType, params string[] otherContentTypes)
+        {
+            ContentTypes = GetContentTypes(contentType, otherContentTypes);
+        }
+
+        public int Order { get; } = Int32.MaxValue - 10;
```

Comment:
Ordering here is important because it determines what policy is applied in what order. HttpMethods for instance are order 100, so an HttpMethodConstraint can't be the tiebreaker if two actions with Consumes attributes match. 

I'd also suggest a lower number here, like 200.